### PR TITLE
Create work term for Pressure Changer with EnergyBalanceType.none

### PR DESCRIPTION
**Pull request by adam-a-a**
## Summary/Motivation:
In WaterTAP, we noticed that having a property model with `EnergyBalanceType.none` would be incompatible with a Pump unit. Thus, to be compatible with the Pump, the property model currently needs to have  `EnergyBalanceType.enthalpyTotal` set. However, enthalpy is not necessary for the Pump; we only really care about volumetric flow rate, pressure (head) and efficiency to then calculate work. 

In short, the `work` variable is only ever created on a ControlVolume Block when `EnergyBalanceType.enthalpyTotal` is set which shouldn't always be the case. For more details, please have a look at this discussion in our WaterTAP repo: https://github.com/watertap-org/watertap/issues/651

## Changes proposed in this PR:
- add conditional check before Reference connecting `control_volume.work` to `self.work_mechanical` in pressure_changer unit
- if the conditional is true, then `work` variable is created on control volume
- the conditional ensures the following is true before creating `work` on control volume within pressure_changer unit:
1. `work` does not exist on control volume
2. `EnergyBalanceType` is none
3. `ThermodynamicAssumption` is `Pump`

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
4. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/models/unit_models/pressure_changer.py [hunk 1]
```diff
@@ -342,6 +343,20 @@ def build(self):
...
         # Set references to balance terms at unit level
         # Add Work transfer variable 'work'
+        # If the 'work' variable wasn't already built on the control volume but is needed, create it now.
+        if (
+            not hasattr(self.control_volume, "work")
+            and self.config.thermodynamic_assumption == ThermodynamicAssumption.pump
+            and eb is None
+        ):
+            units = self.config.property_package.get_metadata().get_derived_units
+            self.control_volume.work = Var(
```
Comment (andrewlee94) [inline idaes/models/unit_models/pressure_changer.py:353]: Should we be adding the `work` term to the control volume in this case, or just create `work_mechanical` as a `Var` at the unit level?

File: idaes/models/unit_models/pressure_changer.py [hunk 2]
```diff
@@ -444,7 +459,7 @@ def add_adiabatic(self):
 
         @self.Constraint(self.flowsheet().time)
         def zero_work_equation(b, t):
-            return self.control_volume.work[t] == 0
+            return b.control_volume.work[t] == 0
```
Comment (andrewlee94) [inline idaes/models/unit_models/pressure_changer.py:462]: I would argue this is the reverse of what you should do - it would probably be better to have `self` as the argument in both cases.

Also, whilst not part of this PR, this option is misnamed. `adiabatic` means no heat transfer (which is true for all our options), not no work transfer.

File: idaes/models/unit_models/tests/test_pressure_changer.py [hunk 3]
```diff
@@ -845,6 +844,38 @@ def test_config(self):
...
+            default={
+                "property_package": m.fs.properties,
+                "energy_balance_type": EnergyBalanceType.none,
+            }
+        )
+
+        assert m.fs.unit.config.energy_balance_type == EnergyBalanceType.none
+        assert hasattr(m.fs.unit.control_volume, "work")
+
+        m2 = ConcreteModel()
```
Comment (andrewlee94) [inline idaes/models/unit_models/tests/test_pressure_changer.py:865]: This looks like it should be two separate tests.

File: idaes/models/unit_models/tests/test_pressure_changer.py [hunk 4]
```diff
@@ -845,6 +844,38 @@ def test_config(self):
...
+
+        m.fs.unit = Pump(
+            default={
+                "property_package": m.fs.properties,
+                "energy_balance_type": EnergyBalanceType.none,
+            }
+        )
+
+        assert m.fs.unit.config.energy_balance_type == EnergyBalanceType.none
+        assert hasattr(m.fs.unit.control_volume, "work")
```
Comment (andrewlee94) [inline idaes/models/unit_models/tests/test_pressure_changer.py:863]: You should probably also check the reference here too.

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
A few minor comments.

One thing not covered here - we should check that the documentation discussed what happens in these cases.

File: idaes/models/unit_models/pressure_changer.py (hunk 2 above)
Comment (adam-a-a) [reply idaes/models/unit_models/pressure_changer.py:462]: I agree. All I did was update `self` to `b` on a whim and has no relevance to the PR, so I can just change it back.

File: idaes/models/unit_models/pressure_changer.py (hunk 1 above)
Comment (adam-a-a) [reply idaes/models/unit_models/pressure_changer.py:353]: I started with that approach, but turned out to be a bit cleaner without cascading changes necessary.

Comment (adam-a-a) [reply idaes/models/unit_models/pressure_changer.py:353]: Left this one open for the time being, subject to other reviewers' thoughts.

Comment (eslickj) [reply idaes/models/unit_models/pressure_changer.py:353]: I'm fine with adding it to the control volume I think.  I suppose it is the expected location for it.

**Review by eslickj — APPROVED** [review]
Looks good as far as I can tell.

**Review by MAZamarripa — APPROVED** [review]
lgtm

**Review by andrewlee94 — APPROVED** [review]

File: idaes/models/unit_models/pressure_changer.py (hunk 1 above)
Comment (lbibl) [reply idaes/models/unit_models/pressure_changer.py:353]: @andrewlee94 ,  I looked over this discussion and understood either way above would do the job well.  But while I was reading a initialization API design proposal you wrote,  I would like to say that I think it would be better to only create `work_mechanical`,  as you suggested, by the principle that a var associated with one block should stay with this block, rather than being placed into another block like what is currently.  This way will also be documented more clearly, but I think the doc PR I just opened described the current version well.    Just a thought.  I would be happy to help change it if the IDAES team decides to, otherwise I look forward to concluding the doc PR I opened.  Thanks.
